### DISCUSSION TITLE
Add a config to enable the NumLock key on login

### DISCUSSION THREAD

**Discussion by stepanzak** [YaLTeR/niri#290] [Ideas]
Hi, I wasn't able to find a simple way to enable the NumLock key on login to Niri session, so I don't have to press NumLock to be able to type on numerical keyboard. In fact, I wasn't able to find any way to do that. Running the `numlockl on` command using [Numlockl](https://gitlab.com/phoepsilonix/numlockl) turns the NumLock on, but running it with spawn-at-startup or as a user systemd service doesn't. Both Hyprland and Sway have a config to enable it, so I think it would be awesome if Niri had it too.

**Comment by YaLTeR** [comment]
I agree; this may or may not need Smithay additions (I'm not sure).

**Comment by dukedook** [comment]
One option for this is to use `options "numpad:mac"`.  The downside is it doesn't allow you to switch back to using the numpad as arrow keys etc.

**Reply by stepanzak** [reply]
Thanks, that's great!
I honestly don't know anyone who uses numpad as arrows, so this is enough for me and probably for most people.

**Reply by andybalholm** [reply]
It would be perfect for me, except that some Java software doesn't respect that keyboard layout setting, so it needs NumLock turned on too. :-(

**Reply by RichardFevrier** [reply]
OMG ur my savior I was going to turn crazy after months of using Niri 😄
